lib/phase/flatten.js

```diff
@@ -105,6 +107,10 @@ function merge(newJob, oldJob, fromTemplate) {
         newJob.provider = oldJob.provider || {};
     }
 
+    if (oldJob.stage) {
```

Comment:
Related to https://github.com/screwdriver-cd/config-parser/pull/143/files#r1242377069

Reply:
I think it's nice/quicker way to know if a job belongs to a stage instead of looking it up every time